apps/dashboard/app/helpers/scripts_helper.rb

```diff
@@ -5,7 +5,9 @@ module ScriptsHelper
   def create_editable_widget(form, attrib, format: nil)
     widget = attrib.widget
     attrib.html_options = { class: 'real-field', autocomplete: 'off' }
-    locals = { form: form, attrib: attrib, format: format }
+    fixed = attrib.fixed?
+    attrib.opts[:fixed] = false
```

Comment:
In this solution, this is required to render fixed form elements in their usual way.

Reply:
I see what's going on here now. We call `create_editable_widget` that calls a given editable partial and within _those editable partials_, we call `create_widget`.

Reply:
This is fine - but I should let you know that I think we need a bit of javascript actually to turn this flip this to readonly even in `edit` views.

The UX I'm thinking of is like this (in the edit page):
* when you click 'static value' the actual thing becomes readonly.
* clicking 'static value' again resets this.

I think it'll help reinforce what the user is doing here.

But that builds on top of what you have - we still want to render a select dropdown or number input (as apposed to a readonly text input) and just flip it's readonly value as you're editing it.